env.go

```diff
@@ -585,16 +599,11 @@ func (e *Env) Resolve(l *ui.UI, selector manifest.Selector, syncOnMissing bool)
 // Returns the resolution errors for versions as warnings.
 // If a version fails to resolve for all systems, returns an error
 func (e *Env) ValidateCompatibility(l *ui.UI, name string) (warnings []string, err error) {
-	type system struct {
-		os   string
-		arch string
-	}
-
-	systems := []system{
-		{"darwin", "arm64"},
-		{"darwin", "amd64"},
-		{"linux", "arm64"},
-		{"linux", "amd64"},
+	platforms := []platform.Platform{
```

Comment:
This should have been replaced with platform.Core?

Reply:
Maybe. I decided not to in this PR as that would had been a change in functionality. Now, if we have a package defined only in `linux-arm64` That can pass the validation. It would be nice to unify these, though.

Reply:
We can decide if we want it here: https://github.com/cashapp/hermit/pull/111